diff -up rssguard-4.0.4/pri/install.pri.orig rssguard-4.0.4/pri/install.pri
--- rssguard-4.0.4/pri/install.pri.orig	2021-10-15 13:13:56.000000000 +0200
+++ rssguard-4.0.4/pri/install.pri	2021-12-18 13:37:57.373830459 +0100
@@ -58,7 +58,7 @@ unix:!mac:!android {
   appdata.path = $$quote($$PREFIX/share/metainfo/)
 
   lib.files = $$OUT_PWD/../librssguard/librssguard.so
-  lib.path = $$quote($$PREFIX/lib/)
+  lib.path = $$quote($$PREFIX/$$LIB_INSTALL_DIR/)
   lib.CONFIG = no_check_exist
 
   desktop_icon.files = ../../resources/graphics/$${TARGET}.png
@@ -115,5 +115,4 @@ for(header, INSTALL_HEADERS) {
 
   eval(headers_$${dirname(header)}.files += $$header)
   eval(headers_$${dirname(header)}.path = $$path)
-  eval(INSTALLS *= headers_$${dirname(header)})
 }
