diff -up rssguard-3.5.7/rssguard.pro.unbundle_qtsinglecoreapplication rssguard-3.5.7/rssguard.pro
--- rssguard-3.5.7/rssguard.pro.unbundle_qtsinglecoreapplication	2019-04-04 09:30:57.000000000 +0200
+++ rssguard-3.5.7/rssguard.pro	2019-04-04 11:23:05.116453958 +0200
@@ -142,6 +142,8 @@ message(rssguard: lrelease executable na
 
 QT *= core gui widgets sql network xml
 
+CONFIG *= qtsingleapplication
+
 CONFIG *= c++1z warn_on
 CONFIG -=  debug_and_release
 DEFINES *= QT_USE_QSTRINGBUILDER QT_USE_FAST_CONCATENATION QT_USE_FAST_OPERATOR_PLUS UNICODE _UNICODE
@@ -307,10 +309,6 @@ HEADERS +=  src/core/feeddownloader.h \
             src/network-web/oauth2service.h \
             src/network-web/silentnetworkaccessmanager.h \
             src/network-web/webfactory.h \
-            src/qtsingleapplication/qtlocalpeer.h \
-            src/qtsingleapplication/qtlockedfile.h \
-            src/qtsingleapplication/qtsingleapplication.h \
-            src/qtsingleapplication/qtsinglecoreapplication.h \
             src/services/abstract/accountcheckmodel.h \
             src/services/abstract/cacheforserviceroot.h \
             src/services/abstract/category.h \
@@ -446,10 +444,6 @@ SOURCES +=  src/core/feeddownloader.cpp
             src/network-web/oauth2service.cpp \
             src/network-web/silentnetworkaccessmanager.cpp \
             src/network-web/webfactory.cpp \
-            src/qtsingleapplication/qtlocalpeer.cpp \
-            src/qtsingleapplication/qtlockedfile.cpp \
-            src/qtsingleapplication/qtsingleapplication.cpp \
-            src/qtsingleapplication/qtsinglecoreapplication.cpp \
             src/services/abstract/accountcheckmodel.cpp \
             src/services/abstract/cacheforserviceroot.cpp \
             src/services/abstract/category.cpp \
diff -up rssguard-3.5.7/src/miscellaneous/application.cpp.unbundle_qtsinglecoreapplication rssguard-3.5.7/src/miscellaneous/application.cpp
--- rssguard-3.5.7/src/miscellaneous/application.cpp.unbundle_qtsinglecoreapplication	2019-04-04 09:30:57.000000000 +0200
+++ rssguard-3.5.7/src/miscellaneous/application.cpp	2019-04-04 11:22:11.202530164 +0200
@@ -397,8 +397,6 @@ void Application::onAboutToQuit() {
 
   // Now, we can check if application should just quit or restart itself.
   if (m_shouldRestart) {
-    finish();
-    qDebug("Killing local peer connection to allow another instance to start.");
 
     // TODO: Start RSS Guard with sleep before it cross-platform way if possible.
     // sleep 5 && "<rssguard-start>".
diff -up rssguard-3.5.7/src/miscellaneous/application.h.unbundle_qtsinglecoreapplication rssguard-3.5.7/src/miscellaneous/application.h
--- rssguard-3.5.7/src/miscellaneous/application.h.unbundle_qtsinglecoreapplication	2019-04-04 09:30:57.000000000 +0200
+++ rssguard-3.5.7/src/miscellaneous/application.h	2019-04-04 11:22:11.202530164 +0200
@@ -3,7 +3,7 @@
 #ifndef APPLICATION_H
 #define APPLICATION_H
 
-#include "qtsingleapplication/qtsingleapplication.h"
+#include <QtSingleApplication>
 
 #include "core/feeddownloader.h"
 #include "definitions/definitions.h"
