diff -up rssguard-3.8.3/pri/install.pri.orig rssguard-3.8.3/pri/install.pri
--- rssguard-3.8.3/pri/install.pri.orig	2020-11-30 08:15:34.000000000 +0100
+++ rssguard-3.8.3/pri/install.pri	2020-12-04 23:26:41.034609152 +0100
@@ -140,10 +140,6 @@ INSTALL_HEADERS = \
 ../librssguard/network-web/urlinterceptor.h \
 ../librssguard/network-web/webfactory.h \
 ../librssguard/network-web/webpage.h \
-../librssguard/qtsingleapplication/qtlocalpeer.h \
-../librssguard/qtsingleapplication/qtlockedfile.h \
-../librssguard/qtsingleapplication/qtsingleapplication.h \
-../librssguard/qtsingleapplication/qtsinglecoreapplication.h \
 ../librssguard/services/abstract/accountcheckmodel.cpp \
 ../librssguard/services/abstract/accountcheckmodel.h \
 ../librssguard/services/abstract/cacheforserviceroot.cpp \
diff -up rssguard-3.8.3/src/librssguard/librssguard.pro.orig rssguard-3.8.3/src/librssguard/librssguard.pro
--- rssguard-3.8.3/src/librssguard/librssguard.pro.orig	2020-11-30 08:15:34.000000000 +0100
+++ rssguard-3.8.3/src/librssguard/librssguard.pro	2020-12-04 23:26:41.034609152 +0100
@@ -23,7 +23,7 @@ message($$MSG_PREFIX: lrelease executabl
 include(../../pri/build_opts.pri)
 
 DEFINES *= RSSGUARD_DLLSPEC=Q_DECL_EXPORT
-CONFIG += unversioned_libname unversioned_soname skip_target_version_ext
+CONFIG += unversioned_libname unversioned_soname skip_target_version_ext qtsingleapplication
 
 RESOURCES += ../../resources/sql.qrc \
              ../../resources/rssguard.qrc
@@ -123,10 +123,6 @@ HEADERS += core/feeddownloader.h \
            network-web/oauth2service.h \
            network-web/silentnetworkaccessmanager.h \
            network-web/webfactory.h \
-           qtsingleapplication/qtlocalpeer.h \
-           qtsingleapplication/qtlockedfile.h \
-           qtsingleapplication/qtsingleapplication.h \
-           qtsingleapplication/qtsinglecoreapplication.h \
            services/abstract/accountcheckmodel.h \
            services/abstract/cacheforserviceroot.h \
            services/abstract/category.h \
@@ -274,10 +270,6 @@ SOURCES += core/feeddownloader.cpp \
            network-web/oauth2service.cpp \
            network-web/silentnetworkaccessmanager.cpp \
            network-web/webfactory.cpp \
-           qtsingleapplication/qtlocalpeer.cpp \
-           qtsingleapplication/qtlockedfile.cpp \
-           qtsingleapplication/qtsingleapplication.cpp \
-           qtsingleapplication/qtsinglecoreapplication.cpp \
            services/abstract/accountcheckmodel.cpp \
            services/abstract/cacheforserviceroot.cpp \
            services/abstract/category.cpp \
diff -up rssguard-3.8.3/src/librssguard/miscellaneous/application.cpp.orig rssguard-3.8.3/src/librssguard/miscellaneous/application.cpp
--- rssguard-3.8.3/src/librssguard/miscellaneous/application.cpp.orig	2020-11-30 08:15:34.000000000 +0100
+++ rssguard-3.8.3/src/librssguard/miscellaneous/application.cpp	2020-12-04 23:26:41.035609132 +0100
@@ -521,8 +521,6 @@ void Application::onAboutToQuit() {
 
   // Now, we can check if application should just quit or restart itself.
   if (m_shouldRestart) {
-    finish();
-    qDebugNN << LOGSEC_CORE << "Killing local peer connection to allow another instance to start.";
 
     if (QProcess::startDetached(QDir::toNativeSeparators(applicationFilePath()), {})) {
       qDebugNN << LOGSEC_CORE << "New application instance was started.";
