diff -up rssguard-3.8.0/pri/install.pri.orig2 rssguard-3.8.0/pri/install.pri
--- rssguard-3.8.0/pri/install.pri.orig2	2020-11-11 12:36:27.104551441 +0100
+++ rssguard-3.8.0/pri/install.pri	2020-11-11 12:37:35.500702201 +0100
@@ -140,10 +140,6 @@ INSTALL_HEADERS = \
 ../librssguard/network-web/urlinterceptor.h \
 ../librssguard/network-web/webfactory.h \
 ../librssguard/network-web/webpage.h \
-../librssguard/qtsingleapplication/qtlocalpeer.h \
-../librssguard/qtsingleapplication/qtlockedfile.h \
-../librssguard/qtsingleapplication/qtsingleapplication.h \
-../librssguard/qtsingleapplication/qtsinglecoreapplication.h \
 ../librssguard/services/abstract/accountcheckmodel.cpp \
 ../librssguard/services/abstract/accountcheckmodel.h \
 ../librssguard/services/abstract/cacheforserviceroot.cpp \
diff -up rssguard-3.8.0/src/librssguard/librssguard.pro.orig2 rssguard-3.8.0/src/librssguard/librssguard.pro
--- rssguard-3.8.0/src/librssguard/librssguard.pro.orig2	2020-11-10 13:21:20.000000000 +0100
+++ rssguard-3.8.0/src/librssguard/librssguard.pro	2020-11-11 12:36:27.104551441 +0100
@@ -23,7 +23,7 @@ message($$MSG_PREFIX: lrelease executabl
 include(../../pri/build_opts.pri)
 
 DEFINES *= RSSGUARD_DLLSPEC=Q_DECL_EXPORT
-CONFIG += unversioned_libname unversioned_soname skip_target_version_ext
+CONFIG += unversioned_libname unversioned_soname skip_target_version_ext qtsingleapplication
 
 RESOURCES += ../../resources/sql.qrc \
              ../../resources/rssguard.qrc
@@ -122,10 +122,6 @@ HEADERS += core/feeddownloader.h \
            network-web/oauth2service.h \
            network-web/silentnetworkaccessmanager.h \
            network-web/webfactory.h \
-           qtsingleapplication/qtlocalpeer.h \
-           qtsingleapplication/qtlockedfile.h \
-           qtsingleapplication/qtsingleapplication.h \
-           qtsingleapplication/qtsinglecoreapplication.h \
            services/abstract/accountcheckmodel.h \
            services/abstract/cacheforserviceroot.h \
            services/abstract/category.h \
@@ -273,10 +269,6 @@ SOURCES += core/feeddownloader.cpp \
            network-web/oauth2service.cpp \
            network-web/silentnetworkaccessmanager.cpp \
            network-web/webfactory.cpp \
-           qtsingleapplication/qtlocalpeer.cpp \
-           qtsingleapplication/qtlockedfile.cpp \
-           qtsingleapplication/qtsingleapplication.cpp \
-           qtsingleapplication/qtsinglecoreapplication.cpp \
            services/abstract/accountcheckmodel.cpp \
            services/abstract/cacheforserviceroot.cpp \
            services/abstract/category.cpp \
diff -up rssguard-3.8.0/src/librssguard/miscellaneous/application.cpp.orig2 rssguard-3.8.0/src/librssguard/miscellaneous/application.cpp
--- rssguard-3.8.0/src/librssguard/miscellaneous/application.cpp.orig2	2020-11-11 12:36:27.105551444 +0100
+++ rssguard-3.8.0/src/librssguard/miscellaneous/application.cpp	2020-11-11 12:39:49.580997782 +0100
@@ -511,8 +511,6 @@ void Application::onAboutToQuit() {
 
   // Now, we can check if application should just quit or restart itself.
   if (m_shouldRestart) {
-    finish();
-    qDebugNN << LOGSEC_CORE << "Killing local peer connection to allow another instance to start.";
 
     if (QProcess::startDetached(QDir::toNativeSeparators(applicationFilePath()), {})) {
       qDebugNN << LOGSEC_CORE << "New application instance was started.";
