diff -up rssguard-3.8.4/src/librssguard/librssguard.pro.orig rssguard-3.8.4/src/librssguard/librssguard.pro
--- rssguard-3.8.4/src/librssguard/librssguard.pro.orig	2021-01-05 14:19:02.000000000 +0100
+++ rssguard-3.8.4/src/librssguard/librssguard.pro	2021-01-08 15:50:46.292334813 +0100
@@ -23,7 +23,7 @@ message($$MSG_PREFIX: lrelease executabl
 include(../../pri/build_opts.pri)
 
 DEFINES *= RSSGUARD_DLLSPEC=Q_DECL_EXPORT
-CONFIG += unversioned_libname unversioned_soname skip_target_version_ext
+CONFIG += unversioned_libname unversioned_soname skip_target_version_ext qtsingleapplication
 
 RESOURCES += ../../resources/sql.qrc \
              ../../resources/rssguard.qrc
@@ -124,10 +124,6 @@ HEADERS += core/feeddownloader.h \
            network-web/oauth2service.h \
            network-web/silentnetworkaccessmanager.h \
            network-web/webfactory.h \
-           qtsingleapplication/qtlocalpeer.h \
-           qtsingleapplication/qtlockedfile.h \
-           qtsingleapplication/qtsingleapplication.h \
-           qtsingleapplication/qtsinglecoreapplication.h \
            services/abstract/accountcheckmodel.h \
            services/abstract/cacheforserviceroot.h \
            services/abstract/category.h \
@@ -279,10 +275,6 @@ SOURCES += core/feeddownloader.cpp \
            network-web/oauth2service.cpp \
            network-web/silentnetworkaccessmanager.cpp \
            network-web/webfactory.cpp \
-           qtsingleapplication/qtlocalpeer.cpp \
-           qtsingleapplication/qtlockedfile.cpp \
-           qtsingleapplication/qtsingleapplication.cpp \
-           qtsingleapplication/qtsinglecoreapplication.cpp \
            services/abstract/accountcheckmodel.cpp \
            services/abstract/cacheforserviceroot.cpp \
            services/abstract/category.cpp \
diff -up rssguard-3.8.4/src/librssguard/miscellaneous/application.cpp.orig rssguard-3.8.4/src/librssguard/miscellaneous/application.cpp
--- rssguard-3.8.4/src/librssguard/miscellaneous/application.cpp.orig	2021-01-05 14:19:02.000000000 +0100
+++ rssguard-3.8.4/src/librssguard/miscellaneous/application.cpp	2021-01-08 15:50:46.292334813 +0100
@@ -532,8 +532,6 @@ void Application::onAboutToQuit() {
 
   // Now, we can check if application should just quit or restart itself.
   if (m_shouldRestart) {
-    finish();
-    qDebugNN << LOGSEC_CORE << "Killing local peer connection to allow another instance to start.";
 
     if (QProcess::startDetached(QDir::toNativeSeparators(applicationFilePath()), {})) {
       qDebugNN << LOGSEC_CORE << "New application instance was started.";
