diff -up rssguard-3.5.4/rssguard.pro.unbundle_qtsinglecoreapplication rssguard-3.5.4/rssguard.pro
--- rssguard-3.5.4/rssguard.pro.unbundle_qtsinglecoreapplication	2017-10-23 06:39:28.000000000 +0200
+++ rssguard-3.5.4/rssguard.pro	2017-11-01 13:18:38.741661979 +0100
@@ -151,6 +151,8 @@ message(rssguard: lrelease executable na
 
 QT *= core gui widgets sql network xml
 
+CONFIG *= qtsingleapplication
+
 CONFIG *= c++11 warn_on
 DEFINES *= QT_USE_QSTRINGBUILDER QT_USE_FAST_CONCATENATION QT_USE_FAST_OPERATOR_PLUS UNICODE _UNICODE
 VERSION = $$APP_VERSION
@@ -282,10 +284,6 @@ HEADERS +=  src/core/feeddownloader.h \
             src/network-web/oauth2service.h \
             src/network-web/silentnetworkaccessmanager.h \
             src/network-web/webfactory.h \
-            src/qtsingleapplication/qtlocalpeer.h \
-            src/qtsingleapplication/qtlockedfile.h \
-            src/qtsingleapplication/qtsingleapplication.h \
-            src/qtsingleapplication/qtsinglecoreapplication.h \
             src/services/abstract/accountcheckmodel.h \
             src/services/abstract/cacheforserviceroot.h \
             src/services/abstract/category.h \
@@ -418,10 +416,6 @@ SOURCES +=  src/core/feeddownloader.cpp
             src/network-web/oauth2service.cpp \
             src/network-web/silentnetworkaccessmanager.cpp \
             src/network-web/webfactory.cpp \
-            src/qtsingleapplication/qtlocalpeer.cpp \
-            src/qtsingleapplication/qtlockedfile.cpp \
-            src/qtsingleapplication/qtsingleapplication.cpp \
-            src/qtsingleapplication/qtsinglecoreapplication.cpp \
             src/services/abstract/accountcheckmodel.cpp \
             src/services/abstract/cacheforserviceroot.cpp \
             src/services/abstract/category.cpp \
diff -up rssguard-3.5.4/src/miscellaneous/application.cpp.unbundle_qtsinglecoreapplication rssguard-3.5.4/src/miscellaneous/application.cpp
--- rssguard-3.5.4/src/miscellaneous/application.cpp.unbundle_qtsinglecoreapplication	2017-10-23 06:39:28.000000000 +0200
+++ rssguard-3.5.4/src/miscellaneous/application.cpp	2017-11-01 13:19:35.593537502 +0100
@@ -397,8 +397,6 @@ void Application::onAboutToQuit() {
 
   // Now, we can check if application should just quit or restart itself.
   if (m_shouldRestart) {
-    finish();
-    qDebug("Killing local peer connection to allow another instance to start.");
 
     // TODO: Start RSS Guard with sleep before it cross-platform way if possible.
     // sleep 5 && "<rssguard-start>".
diff -up rssguard-3.5.4/src/miscellaneous/application.h.unbundle_qtsinglecoreapplication rssguard-3.5.4/src/miscellaneous/application.h
--- rssguard-3.5.4/src/miscellaneous/application.h.unbundle_qtsinglecoreapplication	2017-10-23 06:39:28.000000000 +0200
+++ rssguard-3.5.4/src/miscellaneous/application.h	2017-11-01 11:31:32.199058891 +0100
@@ -3,7 +3,7 @@
 #ifndef APPLICATION_H
 #define APPLICATION_H
 
-#include "qtsingleapplication/qtsingleapplication.h"
+#include <QtSingleApplication>
 
 #include "core/feeddownloader.h"
 #include "definitions/definitions.h"
